### ISSUE TITLE
Public APIs giving 403 in AWS EC2 hosted in Mumbai.

### ISSUE THREAD

**Issue by khushgandhi** [cowinapi/developer.cowin#212]
#19 ,
After including header for "User-Agent" : "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.93 Safari/537.36 Edg/90.0.818.51" I am still getting 403 Error.
![cowin_fail](https://user-images.githubusercontent.com/63594223/118350893-92e23a00-b576-11eb-83df-0ca704f400b3.PNG)

**Comment by monodeepdas1215** [comment]
do you have the credentials ?
If yes can you tell me are you a solo developer and still able to get token ?

**Comment by khushgandhi** [comment]
No @monodeepdas1215 I am using Co-WIN Public APIs which doesn't require any api-key/credentials.
Let me know if I am wrong.

**Comment by sahilt75** [comment]
@khushgandhi Is it because of rate limiting? How many API calls do you make within 5 mins?

**Comment by khushgandhi** [comment]
@sahilt75 yes it is because of rate limiting. I just found out that it is calling 300+ times within 3 mins .
could you please suggest any way to restrict the number of calls.

**Comment by sahilt75** [comment]
@khushgandhi You can do this:
1. Batch your requests into lists of 100 each
2. Schedule each list of requests to be sent at an interval of 5 mins. Keep a buffer of 1-2 mins if you want. 

Hope this helps.

**Comment by AkashKrDutta** [comment]
Can you try this:

```
curl GET "https://cdn-api.co-vin.in/api/v2/appointment/sessions/public/findByDistrict?district_id=168&date=29-05-2021" -H  "accept: application/json" -H  "Accept-Language: en_US" -H "User-Agent: Mozilla/5.0" 2> /dev/null
{"sessions":[]}

```
Works for me. If not, then some firewall/EC2 issue.

**Comment by nikhiltalreja** [comment]
I have an EC2 + Indian IP.. during testing, I accidentally hit the server many times and now I am permanently getting 403. My application is running successfully from local (JIO IP), how can I request them to release my EC2 IP address from their banned list

**Comment by sahilt75** [comment]
@nikhiltalreja Did you try passing custom headers? It doesn't seem like IP issue. You can try another EC2 instance though.

**Comment by nikhiltalreja** [comment]
Few days back moved to a fresh Indian instance on gcp and it got resolved. But anyways, since 24 hours, 403 is hitting all developers.

-Nikhil
